PR: Studio Config Generator

KelSolaar opened the pull request:
This PR implements the Studio config generator on top of the CG and Reference config generators. It relies on #58 being merged first.

I updated the [Studio spreadsheet](https://docs.google.com/spreadsheets/d/1nE95DEVtxtEkcIEaJk0WekyEH0Rcs8z_3fdwUtqP8V4/)  according to the [CG spreadsheet](https://docs.google.com/spreadsheets/d/1DqxmtZpnhL_9N1wayvcW0y3bZoHRom7A1c58YLlr89g/). They are so similar now that I'm wondering if we should not fuse them at some point to lessen maintenance burden.

The currently generated Studio config only include the CG config transforms as I have not included anything else than those in the spreadsheet yet.

doug-walker commented:
There is some overlap with #58, so I'll review this again once that is merged in.

KelSolaar commented:
I have rebased the branch on top of `main` and also added 3 (+3) new tasks to update the config mapping files easily:

- `invoke update-mapping-file-reference`
- `invoke update-mapping-file-cg`
- `invoke update-mapping-file-studio`

KelSolaar commented:
So this is where we are:

# Summary of the Studio Config

## Displays

- sRGB
- Rec.1886 / Rec.709 Video
- Rec.1886 / Rec.2020 Video
- Rec.2100-HLG
- Rec.2100-PQ
- ST2084-P3-D65
- P3-D60
- P3-D65
- P3-DCI

## Linear Colorspaces

- ACES2065-1
- ACEScg
- Linear P3-D65
- Linear Rec.2020
- Linear Rec.709

## Camera & Grading Colorspaces

### ACES

- ACEScc
- ACEScct

### Blackmagic Design

- DaVinci Intermediate WideGamut
- Linear DaVinci WideGamut
- DaVinci Intermediate Log - Curve

### Panasonic

- Linear V-Gamut
- V-Log V-Gamut
- V-Log - Curve

### RED

- Linear REDWideGamutRGB
- Log3G10 REDWideGamutRGB
- Log3G10 - Curve

### Sony

- Linear S-Gamut3
- Linear S-Gamut3.Cine
- Linear Venice S-Gamut3
- Linear Venice S-Gamut3.Cine
- S-Log3 S-Gamut3
- S-Log3 S-Gamut3.Cine
- S-Log3 Venice S-Gamut3
- S-Log3 Venice S-Gamut3.Cine
- S-Log3 - Curve

### ITU

- Camera Rec.709

## Texture Colorspaces

- Gamma 1.8 Rec.709 - Texture
- Gamma 2.2 AP1 - Texture
- Gamma 2.2 Rec.709 - Texture
- Gamma 2.4 Rec.709 - Texture
- sRGB - Texture

## Utility Colorspaces

- Un-tone-mapped
- CIE-XYZ-D65
- Raw
- Rec1886 - Curve
- Rec.709 - Curve
- sRGB - Curve
- ST-2084 - Curve

michdolan commented on opencolorio_config_aces/clf/discover/classify.py:
```diff
@@ -1157,18 +1204,21 @@ def classify_clf_transforms(unclassified_clf_transforms):
+    >>> family = sorted(clf_transforms.keys())[0]
     >>> str(family)
-    'Utility'
+    'blackmagic'
```
Should vendor names be capitalized for family? `Utility` is `utility` in the directory structure, so the `blackmagic` directory should probably be `Blackmagic` for family.

KelSolaar replied:
Yes, so we could do that and it would be fine, the code is actually using the parsed `CLFtransformID` for the config colorspaces which is a bit more solid as the name are expected to be good there! We could make them match that being said.

Shall we do that post Siggraph? There are always shenanigans when trying to change case of files and directories with Git.

michdolan replied:
Sounds good. Post Siggraph is fine.
